soc/xtensa/esp32/Kconfig.soc

```diff
@@ -20,6 +20,22 @@ config ESPTOOLPY_FLASHFREQ_80M
 	bool
 	default y
 
+config FLASH_SIZE
+	int
+	default $(dt_node_reg_size_int,/soc/flash-controller@3ff42000/flash@0,0)
+
+config FLASH_BASE_ADDRESS
+	hex
+	default $(dt_node_reg_addr_hex,/soc/flash-controller@3ff42000/flash@0)
+
+config ESPTOOLPY_FLASHSIZE
+	string
+	default "1MB" if FLASH_SIZE = 0x100000
```

Comment:
@tejlmand This table map is required as input to esptool (CMakeLists.txt). It is also needed for all ESP SoCs, meaning that this will be copied to ESP32-S2 and ESP32-C3 Kconfigs. Is there some common place to put common Espressif Kconfig option considering that both xtensa and riscv architecture are supported?

Reply:
does this really belong in Kconfig ?

Did you coinsider if this isd something that should be calculated inside the build system.

Reply:
@tejlmand This information is used in CMakeLists.txt during esptool call. Better if I move this Kconfig to that place? I'll have to copy and paste the snippet for every Espressif SoC CMakeLists file, correct?